### PR TITLE
add pundit policies for RIDP documents controller

### PR DESCRIPTION
**Pull request by saipraveen18**
# PR Checklist

Please check if your PR fulfills the following requirements
- [X] The title follows our [guidelines](https://github.com/ideacrew/enroll/blob/trunk/CONTRIBUTING.md#commit
)
- [X] Tests for the changes have been added (for bugfixes/features)

# PR Type
What kind of change does this PR introduce?

- [X] Bugfix
- [ ] Feature (requires Feature flag)
- [ ] Data fix or migration (inert code, no impact until run)
- [X] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Dependency updates (e.g., add a new gem or update to a version)

# What is the ticket # detailing the issue?

Ticket: https://www.pivotaltracker.com/story/show/187136950

# A brief description of the changes

Current behavior: RIDP documents controller does not have any pundit policies for controller actions

New behavior: Restrict upload/download/delete of RIDP documents by adding pundit checks in the controller

# Feature Flag

For all new feature development, a feature flag is required to control the exposure of the feature to our end users. A feature flag needs a corresponding environment variable to initialize the state of the flag. Please share the name of the environment variable below that would enable/disable the feature and which client(s) it applies to.

Variable name:

- [ ] DC
- [ ] ME

# Additional Context
Include any additional context that may be relevant to the peer review process.

# AppScan CodeSweep Failure
In the event of a failed check on the AppScan CodeSweep step of our GitHub Actions workflow, please review the False Positive protocol outlined here: appscan_codesweep/CODESWEEP_FALSE_POSITIVES_README.MD

Add all required notes to this section if the failure is a suspected false positive.

### CODE REVIEW

File: app/policies/consumer_role_policy.rb [hunk 1]
```diff
@@ -18,6 +23,45 @@ def privacy?
...
 
+  # Determines if the current user has permission to upload ridp document.
+  # The user can upload the document if they are a primary family member,
+  # an active associated broker, or an admin in the individual market,
+  #
+  # @return [Boolean] Returns true if the user has permission to upload the document, false otherwise.
+  def ridp_document_upload?
+    return true if individual_market_non_ridp_primary_family_member?
+    return true if individual_market_admin?
+    return true if active_associated_individual_market_family_broker?
```
Comment (saikumar9) [inline app/policies/consumer_role_policy.rb:34]: @saipraveen18 Please also include access to the Broker Agency Staff Roles.
- `active_associated_individual_market_family_broker_staff?` or
- `active_associated_individual_market_ridp_verified_family_broker_staff?`

File: app/policies/consumer_role_policy.rb [hunk 2]
```diff
@@ -18,6 +23,45 @@ def privacy?
...
+
+  # Determines if the current user has permission to download ridp document.
+  # The user can download the document if they are a primary family member,
+  # an active associated broker, or an admin in the individual market,
+  #
+  # @return [Boolean] Returns true if the user has permission to download the document, false otherwise.
+  def ridp_document_download?
+    return true if individual_market_non_ridp_primary_family_member?
+    return true if individual_market_admin?
+    return true if active_associated_individual_market_family_broker?
```
Comment (saikumar9) [inline app/policies/consumer_role_policy.rb:47]: @saipraveen18 Please also include access to the Broker Agency Staff Roles.
- `active_associated_individual_market_family_broker_staff?` or
- `active_associated_individual_market_ridp_verified_family_broker_staff?`

File: app/policies/consumer_role_policy.rb [hunk 3]
```diff
@@ -18,6 +23,45 @@ def privacy?
...
+
+  # Determines if the current user has permission to delete ridp document.
+  # The user can delete the document if they are a primary family member,
+  # an active associated broker, or an admin in the individual market,
+  #
+  # @return [Boolean] Returns true if the user has permission to delete the document, false otherwise.
+  def ridp_document_delete?
+    return true if individual_market_non_ridp_primary_family_member?
+    return true if individual_market_admin?
+    return true if active_associated_individual_market_family_broker?
```
Comment (saikumar9) [inline app/policies/consumer_role_policy.rb:60]: @saipraveen18 Please also include access to the Broker Agency Staff Roles.
- `active_associated_individual_market_family_broker_staff?` or
- `active_associated_individual_market_ridp_verified_family_broker_staff?`

**Review by saikumar9 — CHANGES_REQUESTED** [review]

File: app/policies/consumer_role_policy.rb (hunk 1 above)
Comment (saipraveen18) [reply app/policies/consumer_role_policy.rb:35]: added checks for `broker_agency_staff`

File: app/policies/consumer_role_policy.rb (hunk 2 above)
Comment (saipraveen18) [reply app/policies/consumer_role_policy.rb:49]: added checks for `broker_agency_staff`

File: app/policies/consumer_role_policy.rb (hunk 3 above)
Comment (saipraveen18) [reply app/policies/consumer_role_policy.rb:63]: added checks for `broker_agency_staff`

**Review by saikumar9 — APPROVED** [review]
LGTM! Good Job! 👏

**Review by ymhari — APPROVED** [review]

**Pull request merged**